### PR TITLE
TCN-664 automate homebrew-buf release

### PR DESCRIPTION
**Pull request by elliotmjackson**
Automates the two-step process of releasing this asset. 
## `perform`:
```sh
cd "${BUFBUILD_HOMEBREW_BUF_DIR}"
git switch main
git pull origin main

bash update.bash "${VERSION}"
git diff

git checkout -b release/v${VERSION}
git commit -am "Update to v${VERSION}"
git push origin release/v${VERSION}

# Get that PR approved and merged
```

## `update-master`:

```sh
git switch main
git pull origin main
# we keep master around because homebrew cannot deal with the rename
git switch master
git pull origin master
git merge main
git push origin master
```

### CODE REVIEW

**Comment by ElliotMJackson** [comment]
depends on TCN-669

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+jobs:
+  perform:
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.version != '' }}
+    steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: 251311
```
Comment (ElliotMJackson) [inline .github/workflows/release.yaml:23]:
```suggestion
          app_id: 257240
```

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          token: ${{ steps.generate_token.outputs.token }}
+  update-master:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged == true && startsWith(github.event.pull_request.head.ref, 'release')
+    steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: 251311
```
Comment (ElliotMJackson) [inline .github/workflows/release.yaml:54]:
```suggestion
          app_id: 257240
```

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        with:
+          token: ${{ steps.generate_token.outputs.token }}
+          fetch-depth: 0
+      - name: Update master
+        run: |
+          # we keep master around because homebrew cannot deal with the rename
+          git switch master
+          git pull origin master
+          git merge main
+          git push origin master
```
Comment (joshcarp) [inline .github/workflows/release.yaml:75]: is this acceptable?

**Review by joshcarp — APPROVED** [review]
LGTM linking other review which this is basically a copy of: https://github.com/bufbuild/buf/pull/1517#pullrequestreview-1165949665

**Pull request merged**